src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int LENGTH_MULTIPLIER = 2;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] currentHashMap;
+
+    public MyHashMap() {
+        currentHashMap = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
 
+        increaseLength();
+        int bucket = hashCode(key);
+        if (currentHashMap[bucket] == null) {
+            currentHashMap[bucket] = new Node<>(key, value, null);
+            size++;
+        }
+        Node<K, V> newNode = currentHashMap[bucket];
+        while (newNode != null) {
+            if (Objects.equals(newNode.key, key)) {
+                newNode.value = value;
+                return;
+            }
+            if (newNode.next == null) {
+                newNode.next = new Node<>(key, value, null);
+                size++;
+                return;
+            }
+            newNode = newNode.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = currentHashMap[hashCode(key)];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void increaseLength() {
+        if (size >= currentHashMap.length * LOAD_FACTOR) {
```

Comment:
Move IF out of this method or rename the method (to make it clear that we do not always increase) .